opensearch/aryn-opensearch.sh

```diff
@@ -0,0 +1,427 @@
+#!/bin/bash
+# TODO: eric - detect low disk space and error out.
+# on macos you fix it in docker desktop > settings > resources > scroll down > virtual disk limit
+main() {
+    local http_status_port=43477 # from rand
+    local http_server_dir=/tmp/http_server
+    local http_status_file="${http_server_dir}/statusz"
+    [[ ! -f "${http_status_file}" ]] || rm -f "${http_status_file}"
+    
+    BASE_URL=http://localhost:9200
+    ARYN_STATUSDIR=/usr/share/opensearch/data/aryn_status
+    mkdir -p "${ARYN_STATUSDIR}"
+
+    set -e
+    # TODO: eric - figure out how we can get the Aryn logs during startup
+    # and switchover to the opensearch ones once we've got everything running
+
+    LOG_BASE="${ARYN_STATUSDIR}/opensearch.log"
+    if opensearch_up; then
+        echo "OpenSearch appears to already be running, not starting it again"
+    else 
```

Comment:
not sure if this would useful, but docker supports healthchecks - https://docs.docker.com/engine/reference/builder/#healthcheck

Reply:
I can't find any way to check the health from another container, so I don't think it's useful to the automation scripts.  I searched for ~5 min, resolve the conversation if you can't find a way to support cross-container healthchecks.

In this particular case, this part of the code supports running opensearch in the container and then logging into the container and running the script again to redo the configuration. Starting opensearch twice wouldn't work well in that case.

Reply:
I was playing around with this the other day.

https://gist.github.com/austintlee/84d24b040992127512e60877d48b8f54

Reply:
Per discussion, this would let us sequence containers with depends_on, but we don't really want to do that.